server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -257,6 +262,19 @@ static void ensureCompatibleVersion(
         Object channel
     ) {
         if (TransportVersion.isCompatible(remoteTransportVersion)) {
+            if (remoteTransportVersion.before(V8_18_HANDSHAKE_VERSION)) {
+                deprecationLogger.warn(
+                    DeprecationCategory.OTHER,
+                    "handshake_version",
+                    "Performed a handshake with a remote node whose version will be incompatible after this node "
+                        + "is upgraded to 9.x. This is likely a result of a CCR/CCS request. "
```

Comment:
I don't think that's right, it could just as well be because there's an ongoing upgrade.